components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/service/impl/HTTPEventPublisherImpl.java

```diff
@@ -84,72 +96,105 @@ public boolean canHandleEvent(EventContext eventContext) throws EventPublisherEx
         }
     }
 
-    private void makeAsyncAPICall(SecurityEventTokenPayload eventPayload, EventContext eventContext) {
+    private void makeAsyncAPICall(SecurityEventTokenPayload eventPayload, EventContext eventContext)
+            throws WebhookMgtException {
+
+        // Freeze immutable per-publish values; reuse across retries.
+        final String correlationId = HTTPAdapterUtil.getCorrelationID(eventPayload);
+        final String tenantDomain = eventContext.getTenantDomain() + "-" + UUID.randomUUID();
```

Comment:
Why are we appending random uuid to the tenant domain ?

Reply:
Fixed via https://github.com/wso2-extensions/identity-event-publishers/pull/47/commits/716e45976f7a7266efef45f17d292a4ca936da4c